### DISCUSSION TITLE
Why are only four numbers recognized in NumberFormat?

### DISCUSSION THREAD

**Discussion by REDLIGHTFILMS** [react-hook-form/react-hook-form#4506] [Q&A]
Hi, I have the following problem, when I enter a value in my TextField, which is embedded in NumberFormat. Then only three numbers are accepted and then the field is reset. I want to divide my value in my textfield by a format, this is determined by the drop down menu. So if I select the value thousand then the value in the text field should be calculated by 1000. But as soon as I divide the value in the text field by the number, from the drop down menu, then I can only enter three numbers....
How to solve this problem, unfortunately I have no approach at all?

Example Code: https://codesandbox.io/s/hopeful-babbage-vrhfx?file=/src/App.tsx

**Comment by patrickap** [comment]
Hi that doesn't seem to be due to react hook form but to the formatting (thousand and decimal separators) of the number format component. After removing the props it will work.

**Reply by REDLIGHTFILMS** [reply]
How exactly do you mean that I should remove the props?.

**Reply by patrickap** [reply]
Remove these lines and test again. It will work:

```javascript
...
thousandSeparator="."
decimalSeparator=","
...
```

Your value results to `NaN` after dividing using this format configuration.
